pkl-core/src/main/java/org/pkl/core/stdlib/base/StringNodes.java

```diff
@@ -739,6 +739,21 @@ protected VmList eval(String self, VmRegex separator) {
     }
   }
 
+  public abstract static class splitLimit extends ExternalMethod2Node {
+    @TruffleBoundary
+    @Specialization
+    protected VmList eval(String self, String separator, long limit) {
+      var parts = self.split(Pattern.quote(separator), (int) limit);
+      return VmList.create(parts);
+    }
+
+    @TruffleBoundary
+    @Specialization
+    protected VmList eval(String self, VmRegex separator, long limit) {
+      return VmList.create(separator.getPattern().split(self), (int) limit);
```

Comment:
It took me far too long to find this, but considering my comment on your snippet test expectation, I _believe_ this should be...
```suggestion
      return VmList.create(separator.getPattern().split(self, (int) limit));
```
I don't mind either way, but the difference would have been easier to spot if this was written in the same form as `eval(String,String,long)`;
```
var parts = separator.getPattern().split(self, (int) limit);
return VmList.create(parts);
```

Reply:
Oof good catch. This is fixed now.